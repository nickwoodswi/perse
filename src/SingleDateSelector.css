/* see RangeDateSelector.css for applicable styling */