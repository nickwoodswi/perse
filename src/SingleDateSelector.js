import React, { Component } from 'react'
import MonthDayCounts from './MonthDayCounts.js'
// see RangeDateSelector.css for applicable styling

class SingleDateSelector extends Component{
    render() {
        const months = Object.keys(MonthDayCounts)
        const days = MonthDayCounts[this.props.selectedMonth]
        return(
            <>
            <div className="date-selector">
                <label htmlFor="start-month-selector">Select Month</label>
                <select 
                    id="start-month-selector"
                    key="start-month-selector" 
                    className="month-selector" 
                    onChange={e => this.props.handleSingleDate('month_start', 'month_end', e.target.value)}>
                        {months.map((month, idx) => {
                            return(
                                <option key={idx}>{month}</option>
                            )
                        })}
                </select>
                <label htmlFor="start-day-selector">Select Start Day</label>
                <select 
                    id="start-day-selector"
                    key="start-day-selector" 
                    className="day-selector" 
                    onChange={e => this.props.handleSingleDate('day_start', 'day_end', e.target.value)}>
                    <>{days}</>
                </select>
                <label htmlFor="start-year-selector">Select Start Year</label>
                <select 
                    id="start-year-selector"
                    key="start-year-selector" 
                    className="year-selector" 
                    onChange={e => this.props.handleSingleDate('year_start', 'year_end', e.target.value)}>
                        <option>2020</option>
                        <option>2021</option>
                </select>
            </div>
            </>
        )
    }
}

export default SingleDateSelector