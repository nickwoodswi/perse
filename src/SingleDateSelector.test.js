import React from 'react';
import { render } from '@testing-library/react';
import SingleDateSelector from './SingleDateSelector';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders single date selector', () => {
    render(<MemoryRouter><SingleDateSelector /></MemoryRouter>);
});

it('SingleDateSelector renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><SingleDateSelector /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});