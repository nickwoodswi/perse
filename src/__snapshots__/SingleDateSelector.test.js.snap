// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDateSelector renders the UI as expected 1`] = `
<div
  className="date-selector"
>
  <label
    htmlFor="start-month-selector"
  >
    Select Month
  </label>
  <select
    className="month-selector"
    id="start-month-selector"
    onChange={[Function]}
  >
    <option>
      MONTH
    </option>
    <option>
      January
    </option>
    <option>
      February
    </option>
    <option>
      March
    </option>
    <option>
      April
    </option>
    <option>
      May
    </option>
    <option>
      June
    </option>
    <option>
      July
    </option>
    <option>
      August
    </option>
    <option>
      September
    </option>
    <option>
      October
    </option>
    <option>
      November
    </option>
    <option>
      December
    </option>
  </select>
  <label
    htmlFor="start-day-selector"
  >
    Select Start Day
  </label>
  <select
    className="day-selector"
    id="start-day-selector"
    onChange={[Function]}
  />
  <label
    htmlFor="start-year-selector"
  >
    Select Start Year
  </label>
  <select
    className="year-selector"
    id="start-year-selector"
    onChange={[Function]}
  >
    <option>
      2020
    </option>
    <option>
      2021
    </option>
  </select>
</div>
`;
